model_analyzer/analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -48,75 +48,74 @@ def __init__(self, config):
-
-        Parameters
-        ----------
-        sweep_params : dictionary of lists
-            keys are arguments to perf_analyzer, values are
-            list of argument values to run perf_analyzer with.
+        Runs the perf analyzer with the
+        intialized configuration
 
         Returns
         -------
-        List of tuples
-            (params, output) where params are the set of
-            parameters provided for that run and output is
-            the PerfRecord for the perf_analyzer run
-
-        Raises
-        ------
-        TritonModelAnalyzerException
-            If _run_perf_analyzer throws this exception
-        """
-        # Create a config for each combination of parameters
-        combos = list(product(*tuple(sweep_params.values())))
-        run_configs = [dict(zip(sweep_params.keys(), vals)) for vals in combos]
-
-        # Collect a tuple (run_config, output)
-        run_outputs = []
-
-        # Run parameters are some subset of perf_analyzer config
-        for run_config in run_configs:
-
-            # update perf_analyzer config
-            for key, val in run_config.items():
-                self._config[key] = val
-
-            # Run perf_analyzer
-            run_record = self._run_perf_analyzer()
-            run_outputs.append((run_config, run_record))
-
-        return run_outputs
-
-    def _run_perf_analyzer(self):
-        """
-        Runs the perf_analyzer tool locally
-        and synchronously
-
-        Returns
-        -------
-        PerfRecord
-            output from the subprocess call to
-            the perf_analyzer
+        List of Records for metrics
+        that are generated by the perf analyzer
 
         Raises
         ------
         TritonModelAnalyzerException
             If subprocess throws CalledProcessError
         """
+
         cmd = ['perf_analyzer']
         cmd += self._config.to_cli_string().replace('=', ' ').split()
 
         # Synchronously start and finish run
-
         try:
             out = check_output(cmd, stderr=STDOUT, encoding='utf-8')
         except CalledProcessError as e:
             raise TritonModelAnalyzerException(
                 f"perf analyzer returned with exit"
                 " status {e.returncode} : {e.output}")
 
-        return PerfRecord(out)
+        return self._parse_perf_output(out)
+
+    def _parse_perf_output(self, perf_out):
+        """
+        Extracts required metrics from
+        perf_analyzer output and sets
+        properties for this class
+
+        Parameters
+        ----------
+        perf_out : str
+            The raw output from the subprocess
+            call to perf_analyzer
+        """
+        data = {}
+        # Split output into lines and ignore last 3 lines
+        perf_out_lines = perf_out.split('\n')
+        for line in perf_out_lines[:-3]:
+            # Last word in line with Batch and concurrency
+            if 'Batch' in line:
```

Comment:
There needs to be a unit test to test every one of this if/else conditions. I didn't see this.